free-1.md

```diff
@@ -10,41 +10,148 @@ _What need is met by your idea? Who are you helping? What is that person's
+
+The first is a type of software called a Digital Audio Workstation (DAW).
+Popular DAWs like Pro Tools and FL Studio usually have built-in digital
+"instruments" which user's can program by writing a MIDI score. These are
+usually prefabricated instruments that can only be customized by changing
+certain parameters. This makes it possible to get a range of sounds out of each
+instrument, but it is hard or impossible to create completely new instruments.
+
+Musicians interested in creating their own instruments can use a DSL called
+CSound (which is taught at Mudd in MUS 80). CSound allows users to create
+chains of oscillators -- primitive components used to generate a sine wave --
+and effects like filters to literally build from scratch their own digital
+instruments. This simulates the way the earliest synthesizers were used; the
+musician would connect oscillators and other components using wires. CSound
+takes this process and makes it digital.
+
+But there's a problem with this current state of things: DAWs operate at a high
+level, where the primitive unit is an instrument. CSound operates at a much
+lower level, where the primitive unit is an oscillator (this fact and its
+opcode-based syntax means it is used almost exclusively by academics). There is
+a lack of tools in the middle. Musicians who want to create their own
+instruments but don't want to deal with the low level of abstraction offered by
+CSound have little recourse.
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+The domain of instrument creation has very clearly defined primitives at the
+level of abstraction I'm interested in. I will call these primitives sources
+and effects.
+
+A source is a sound generator, like a CSound oscillator. But a source generates
+a more complex signal than an oscillator, and is in fact a particular
+arrangement of many oscillators. A source is a pattern that can be described by
+a few parameters, but that would have to be implemented over and over again
+from the oscillator level in CSound. With only a few types of primitive sources
+-- additive and subtractive synthesizers, frequency and amplitude modulation
+synths, and physical models -- a wide variety of sounds can be created.
+
+The second type of primitive is an effect. These do not produce sound, but
+rather take a signal as input and output a different, transformed signal. In
+CSound, effects must be implemented from filters and envelopes. At a more
+intuitive (for musicians) level of abstraction, primitive effects would include
+compressors and equalizers. These effects could be chained together with
+sources to alter the sound produced by the sources.
+
+A good tool for digital music synthesis should provide these primitives, and
+should allow them to be composed by chaining in serial and in parallel. This
+already sounds like a language -- we have vocabulary (sources and effects) and
+fluency (the method for chaining).
```

Comment:
I think this breakdown into these two primitives is a good possibility for the language. I wonder if people need to know any background information about oscillators, filters, compressors, etc. in order to appropriately use this DSL.